File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int mx = root.val;
        int level = 1;
        int temp = 0;
        while(q.size()!=0){
            int sz = q.size();
            temp++;
            int sum = 0;
            while(sz-->0){
                TreeNode front = q.remove();
                sum+= front.val;
                if(front.left!=null) q.add(front.left);
                if(front.right!=null) q.add(front.right);
                // sz--;
            }
            if(sum>mx){
                mx = Math.max(mx, sum);
                level = temp;
            }
        }
        return level;
    }
}

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        int mx = root->val;
        int ans_level = 1;
        int level = 0;
        while(q.size()){
            int sz = q.size();
            int sum = 0;
            level++;
            while(sz--){
                TreeNode* top = q.front();
                q.pop();
                sum+=top->val;

                if(top->left!=NULL)q.push(top->left);
                if(top->right!=NULL)q.push(top->right);
            }
            cout<<sum<<" "<<level;
            if(sum>mx){
                mx = max(mx, sum);
                ans_level = level;
            }
        }
        return ans_level;
    }
};